ams/AssetAnalyzer.cs

```diff
@@ -30,16 +31,40 @@ internal class AssetAnalyzer : BaseMigrator
             _tracker = tracker;
         }
 
-        private async Task<AnalysisResult> AnalyzeAsync(MediaAssetResource asset, BlobServiceClient storage, CancellationToken cancellationToken)
+        private async Task<AnalysisResult> AnalyzeAsync<T>(T item, BlobServiceClient storage, CancellationToken cancellationToken)//MediaAssetResource,BlobContainerItem
```

Comment:
You may need to sync the source tree from the main branch and merge with your change, some new changes in main branch would have some impacts on your new change.